projects/step-frontend/src/lib/modules/execution/components/alt-execution-progress/alt-execution-progress.component.html

```diff
@@ -60,6 +60,17 @@ <h3 data-step-testid>
           </step-popover-content>
         </step-popover>
       }
+      @if (agentsInvolved$ | async; as agentsInvolved) {
+        <button
+          type="button"
+          mat-icon-button
+          color="default"
+          (click)="_dialogs.openAgentsModal(agentsInvolved, execution?.description || 'Unknown plan')"
+          matTooltip="Show agents"
+        >
+          <step-icon name="agent" />
+        </button>
+      }
```

Comment:
I think observable can be omitted, because `agentsInvoled` is execution's field.
Then it can be removed from TS code also
```suggestion
      @if (execution?.agentsInvolved; as agentsInvolved) {
        <button
          type="button"
          mat-icon-button
          color="default"
          (click)="_dialogs.openAgentsModal(agentsInvolved, execution?.description || 'Unknown plan')"
          matTooltip="Show agents"
        >
          <step-icon name="agent" />
        </button>
      }
```